reactor/reactor-netty
# Migrate documentation to Antora

This PR is first attempt to migrate Reactor Netty 1.1.x documentation to Antora 

there are three logical commits in this PR:

- 5d55948ffc429a87328ca53cdfeee50b6bb820a3: it contains all modifs done in .adoc files, and also contains antora yaml files
-  5d57f4dac9e9e7dc603b7feca73cddc97a1fc74e: contains all adaptation done in various gradle scripts
- 47cd990f1ab20d96b52152aa74ee3fcdcc20ef90: contains adaptations done in the GH worfklows.


To test this PR, you can proceed like the following:

### Testing without PDF:

- clone this PR
- switch to java 17 (the docs sub project requires java17)
- build the doc and generate it to docs/build/site/index.html:
```
./gradlew antora
```
- you can now browse docs/build/site/index.html
- switch to java 8
- deploy reactor-netty docs to your local M2 
```
./gradlew publishToMavenLocal
```
- now, you can verify the content of ~/.m2/repository/io/projectreactor/netty/reactor-netty/1.1.18-SNAPSHOT/reactor-netty-1.1.18-SNAPSHOT-docs.zip

### Testing with PDF:

as before, PDF are not generated by default in snapshot mode. If you want to also include a PDF in the generated docs zip, then do the following:

- first, you will need to install **asciidoctor-pdf** command, because the antora pdf assembler requires it (this command is provided when doing "brew install asciidoctor"):
```
brew install asciidoctor
```
- switch to java17
- build antora and force PDF generation (in release mode, no need to use -PforcePdf, it's like before):
```
./gradlew antora -PforcePdf
this will generate  ./docs/build/assembler/reactor-netty/1.1.18/reactor-netty-reference-guide.pdf
```
- switch to java 8
- deploy reactor-netty docs to your local M2:
```
./gradlew publishToMavenLocal -PforcePdf
```
- verify the content of ~/.m2/repository/io/projectreactor/netty/reactor-netty/1.1.18-SNAPSHOT/reactor-netty-1.1.18-SNAPSHOT-docs.zip, which should also include a pdf in the root dir of the zip.
